src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionGradleIntegrationTest.groovy

```diff
@@ -111,6 +111,63 @@ class BuildScanInjectionGradleIntegrationTest extends BaseGradleIntegrationTest
         System.clearProperty(TimestampNote.systemProperty)
     }
 
+    def "captures build agent errors in pipeline build if checking for errors is enabled"() {
+        given:
+        def gradleVersion = '8.1.1'
+        gradleInstallationRule.gradleVersion = gradleVersion
+        gradleInstallationRule.addInstallation()
+
+        DumbSlave agent = createSlave()
+
+        def pipelineJob = j.createProject(WorkflowJob)
+
+        pipelineJob.setDefinition(new CpsFlowDefinition("""
+    stage('Build') {
+      node('foo') {
+        withGradle {
+          def gradleHome = tool name: '${gradleInstallationRule.gradleVersion}', type: 'gradle'
+          writeFile file: 'settings.gradle', text: ''
+          writeFile file: 'build.gradle', text: ""
+          if (isUnix()) {
+            sh "'\${gradleHome}/bin/gradle' help --no-daemon --console=plain -Dcom.gradle.scan.trigger-synthetic-error=true"
+          } else {
+            bat(/"\${gradleHome}\\bin\\gradle.bat" help --no-daemon --console=plain -Dcom.gradle.scan.trigger-synthetic-error=true/)
+          }
+        }
+      }
+    }
+""", false))
+
+
+        when:
+        // first build to download Gradle
+        def build = j.buildAndAssertSuccess(pipelineJob)
+
+        then:
+        j.assertLogNotContains(MSG_INIT_SCRIPT_APPLIED, build)
+
+        when:
+        // first build to download Gradle
+        def firstRun = j.buildAndAssertSuccess(pipelineJob)
+
+        then:
+        j.assertLogNotContains(MSG_INIT_SCRIPT_APPLIED, firstRun)
+
+        when:
+        enableBuildInjection(agent, gradleVersion)
+        withInjectionConfig {
+            checkForBuildAgentErrors = true
+        }
```

Comment:
I think you should either swap lines `157` and `158` places, or restart a slave agent to propagate the changes. The `withInjectionConfig` doesn't restart it, so I don't this the change was applied. Does the assertion below really work for you?

Reply:
🤔 Why a restart would be required ? This flag is only read in memory on master, it does not need propagation to agents (and even if it was, we should not need to restart thanks `InjectionConfigChangeListener`)

Reply:
So I ran this test locally and this last assertion fails for me. 
The `InjectionConfigChangeListener` would definitely do the job (I've swapped the lines and it didn't help either, so I'm guessing the problem is somewhere else)